packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -470,7 +470,7 @@ export const getLinksSupplier =
 
       if (link.onClick) {
         linkModel = {
-          href: link.url,
+          href: replaceVariables(link.url, dataLinkScopedVars, VariableFormatID.UriEncode),
```

Comment:
Hey @gng0 👋🏾, thanks for contributing to this change 🙌🏾 . Can you refactor this part to contain the same href processing we do in the `else` condition? 

```
.....
  if (href) {
    href = locationUtil.assureBaseUrl(href.replace(/\n/g, ''));
    href = replaceVariables(href, dataLinkScopedVars, VariableFormatID.UriEncode);
    href = locationUtil.processUrl(href);
```

Reply:
Thanks for catching that @axelavargas 
I've just made the change!